programs/stake/src/stake_state.rs

```diff
@@ -1158,6 +1144,135 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+enum MergeKind {
+    Inactive(Meta, u64),
+    ActivationEpoch(Meta, Stake),
+    FullyActive(Meta, Stake),
+}
+
+impl MergeKind {
+    fn meta(&self) -> &Meta {
+        match self {
+            Self::Inactive(meta, _) => meta,
+            Self::ActivationEpoch(meta, _) => meta,
+            Self::FullyActive(meta, _) => meta,
+        }
+    }
+
+    fn stake(&self) -> Option<&Stake> {
```

Comment:
well, this pr is getting easier to follow ever then ever.

now that we have `active_delegations_can_merge` and `active_stakes_can_merge`, I think it's not too wrong to qualify this accessor fn with `active_`, so rename from `fn stake(...)` => to `fn active_stake(...)`.